dotnet/src/Connectors/Connectors.OpenAIV2/Core/ClientCore.AudioToText.cs

```diff
@@ -0,0 +1,108 @@
+    {
+        if (!input.CanRead)
+        {
+            throw new ArgumentException("The input audio content is not readable.", nameof(input));
+        }
+
+        OpenAIAudioToTextExecutionSettings? audioExecutionSettings = OpenAIAudioToTextExecutionSettings.FromExecutionSettings(executionSettings);
+        AudioTranscriptionOptions? audioOptions = AudioOptionsFromExecutionSettings(executionSettings);
+
+        Verify.ValidFilename(audioExecutionSettings?.Filename);
+
+        var memoryStream = new MemoryStream(input.Data!.Value.ToArray());
+
+        AudioTranscription responseData = (await RunRequestAsync(() => this.Client.GetAudioClient(this.ModelId).TranscribeAudioAsync(memoryStream, audioExecutionSettings?.Filename, audioOptions)).ConfigureAwait(false)).Value;
+
+        return [new(responseData.Text, this.ModelId, metadata: GetResponseMetadata(responseData))];
+    }
+
+    /// <summary>
+    /// Converts <see cref="PromptExecutionSettings"/> to derived <see cref="OpenAIAudioToTextExecutionSettings"/> type.
+    /// </summary>
+    /// <param name="executionSettings">Instance of <see cref="PromptExecutionSettings"/>.</param>
+    /// <returns>Instance of <see cref="OpenAIAudioToTextExecutionSettings"/>.</returns>
+    private static AudioTranscriptionOptions? AudioOptionsFromExecutionSettings(PromptExecutionSettings? executionSettings)
+    {
+        if (executionSettings is null)
+        {
+            return new AudioTranscriptionOptions();
+        }
+
+        if (executionSettings is OpenAIAudioToTextExecutionSettings settings)
+        {
+            return new AudioTranscriptionOptions
+            {
+                Granularities = ConvertToAudioTimestampGranularities(settings.Granularities),
+                Language = settings.Language,
+                Prompt = settings.Prompt,
+                Temperature = settings.Temperature
+            };
+        }
+
+        var json = JsonSerializer.Serialize(executionSettings);
+
+        var openAIExecutionSettings = JsonSerializer.Deserialize<OpenAIAudioToTextExecutionSettings>(json, JsonOptionsCache.ReadPermissive);
+
+        if (openAIExecutionSettings is not null)
+        {
+            return new AudioTranscriptionOptions
+            {
+                Granularities = ConvertToAudioTimestampGranularities(openAIExecutionSettings.Granularities),
+                Language = openAIExecutionSettings.Language,
+                Prompt = openAIExecutionSettings.Prompt,
+                Temperature = openAIExecutionSettings.Temperature
+            };
+        }
+
+        throw new ArgumentException($"Invalid execution settings, cannot convert to {nameof(OpenAIAudioToTextExecutionSettings)}", nameof(executionSettings));
+    }
+
+    private static AudioTimestampGranularities ConvertToAudioTimestampGranularities(IEnumerable<OpenAIAudioToTextExecutionSettings.TimeStampGranularities>? granularity)
+    {
+        return granularity?.FirstOrDefault() switch
```

Comment:
The execution settings allow for supplying more than one granularity, while this code only uses the first one. Is that intentional?

Reply:
Answered above, I'm updating the code to be single granularity